packages/victory-core/src/victory-label/victory-label.js

```diff
@@ -110,36 +115,225 @@ const getTransform = (props) => {
+
+  if (direction === "rtl") {
+    return x - labelSizeWidth;
+  }
+
+  switch (textAnchor) {
+    case "start":
+      return x;
+    case "middle":
+      return Math.round(x - labelSizeWidth / 2);
+    case "end":
+      return Math.round(x - labelSizeWidth);
+    default:
+      return x;
+  }
+};
+
+const getYCoordinate = (calculatedProps, props, textHeight) => {
+  const { verticalAnchor, y } = calculatedProps;
+  const { dy, inline } = props;
+  const offset = y + (dy || 0);
+
+  switch (verticalAnchor) {
+    case "start":
+      return Math.floor(offset);
+    case "middle":
+      return Math.floor(offset - textHeight / 2);
+    case "end":
+      return inline ? Math.floor(offset) : Math.ceil(offset - textHeight);
+    default:
+      return inline ? Math.floor(offset) : Math.floor(offset - textHeight / 2);
+  }
+};
+
+const getBlockTextHeight = (props, adjustedLineHeight, capHeightsPx) => {
+  const { text, style } = props;
+  const styledFontHeight = sumBy(style, (s) => s.fontSize);
+
+  return text.length > style.length
+    ? styledFontHeight * adjustedLineHeight +
+        capHeightsPx +
+        defaultStyles.fontSize * adjustedLineHeight * (text.length - style.length)
+    : styledFontHeight * adjustedLineHeight + capHeightsPx;
+};
+
+const getFullBackground = (props, calculatedProps) => {
+  const { backgroundComponent, backgroundStyle, capHeight, inline, style, text } = props;
+  const { dx, lineHeight, padding } = calculatedProps;
+  const maxString = text.reduce((a, b) => (a.length > b.length ? a : b));
+  const maxFontSize = maxBy(style, (s) => s.fontSize).fontSize;
+  const adjustedLineHeight = checkLineHeight(lineHeight, lineHeight[0], 1);
+  const capHeightsPx = TextSize.convertLengthToPixels(`${capHeight}em`, maxFontSize);
+  const textHeight = inline
+    ? maxFontSize * adjustedLineHeight + capHeightsPx
+    : getBlockTextHeight(props, adjustedLineHeight, capHeightsPx);
+  const width = inline
+    ? TextSize.approximateTextSize(text.join(" "), style).width + (dx || 0) * text.length
+    : TextSize.approximateTextSize(maxString, style).width + (dx || 0);
+  const xCoordinate = getXCoordinate(calculatedProps, width);
+  const yCoordinate = getYCoordinate(calculatedProps, props, textHeight);
   const transform = getTransform(props);
-  const x = props.x !== undefined ? props.x : getPosition(props, "x");
-  const y = props.y !== undefined ? props.y : getPosition(props, "y");
 
-  const textChildren = text.map((line, i) => {
+  const backgroundProps = {
+    height: textHeight + padding.top + padding.bottom,
+    style: backgroundStyle,
+    transform,
+    width: width + padding.left + padding.right,
+    x: xCoordinate,
+    y: yCoordinate
+  };
+
+  return React.cloneElement(
+    backgroundComponent,
+    defaults({}, backgroundComponent.props, backgroundProps)
+  );
+};
+
+const getPaddingProp = (i, padding) => {
+  if (Array.isArray(padding)) {
+    if (padding[i]) {
```

Comment:
If padding is an array, let's have it use `padding[0]` if `padding[i]` doesn't exist to be consistent with how style arrays work